Implement swagger for KEB (#281)
kyma-project/control-plane: docs/kyma-environment-broker/08-07-swagger.md

---
title: Check API using swagger

> type: Tutorials

Reviewer: I'm not sure Tutorials is the right type here since it's not really a step-by-step instruction 🤔 I'd rather classify this as a Details doc.

---

This tutorial shows how to check the API schema using Swagger.

> The Swagger is injected into the application and it's exposes the Swagger UI on the `/swagger/` endpoint.

Reviewer 2: the endpoint has been changed

The Swagger UI static files are copied from the [official source](https://github.com/swagger-api/swagger-ui/tree/master/dist).

> The KEB uses swagger schema file mounted as volume to the Pod. You can find that schema [here](https://github.com/kyma-project/control-plane/blob/master/resources/kcp/charts/kyma-environment-broker/files/swagger.yaml).

Reviewer: These sentences seem to be a bit out of context 🤔 Let's provide some general context to them.

You can use several ways to expose and use the Swagger UI, but two of them are recommended.

## Use Virtual Service

Open the following website:

   ```
   https://$BROKER_URL/
   ```

> > **NOTE:** Using this option you can't use `Try it out` option due to not configured OAuth2 swagger schema.

Reviewer: ```suggestion
> **NOTE:** If you choose this option, you can't use the `Try it out` feature as the OAuth2 Swagger schema is not configured.
```
